src/platform/backends/lxd/lxd_request.cpp

```diff
@@ -82,29 +80,39 @@ const QJsonObject lxd_request_common(const std::string& method, QUrl& url, int t
     if (reply->error() == QNetworkReply::ContentNotFoundError)
         throw mp::LXDNotFoundException();
 
-    if (reply->error() != QNetworkReply::NoError)
-        throw std::runtime_error(fmt::format("{}: {}", url.toString(), reply->errorString()));
+    if (reply->error() == QNetworkReply::OperationCanceledError)
+        throw mp::LXDRuntimeError(
+            fmt::format("Timeout getting response for {} operation on {}", method, url.toString()));
 
     auto bytearray_reply = reply->readAll();
     reply->deleteLater();
 
+    if (bytearray_reply.isEmpty())
+        throw mp::LXDRuntimeError(fmt::format("Empty reply received for {} operation on {}", method, url.toString()));
+
     QJsonParseError json_error;
     auto json_reply = QJsonDocument::fromJson(bytearray_reply, &json_error);
 
     if (json_error.error != QJsonParseError::NoError)
-        throw std::runtime_error(fmt::format("{}: {}", url.toString(), json_error.errorString()));
+        throw mp::LXDRuntimeError(fmt::format("Error parsing JSON response for {}: {}\n{}", url.toString(),
+                                              json_error.errorString(), bytearray_reply));
 
     if (json_reply.isNull() || !json_reply.isObject())
-        throw std::runtime_error(fmt::format("Invalid LXD response for url {}: {}", url.toString(), bytearray_reply));
+        throw mp::LXDRuntimeError(fmt::format("Invalid LXD response for {}: {}", url.toString(), bytearray_reply));
 
     mpl::log(mpl::Level::trace, request_category, fmt::format("Got reply: {}", QJsonDocument(json_reply).toJson()));
 
+    if (reply->error() != QNetworkReply::NoError)
+        throw mp::LXDRuntimeError(fmt::format("Network error for {}: {} - {}", url.toString(), reply->errorString(),
+                                              json_reply.object()["error"].toString()));
+
     return json_reply.object();
 }
 } // namespace
 
 const QJsonObject mp::lxd_request(mp::NetworkAccessManager* manager, const std::string& method, QUrl url,
-                                  const mp::optional<QJsonObject>& json_data, int timeout)
+                                  const mp::optional<QJsonObject>& json_data, int timeout) // clang-format off
+try // clang-format on
```

Comment:
Why these? Do we need to move to newer clang to avoid?

Reply:
I did this because it's used elsewhere in many parts of the code, so I thought it was necessary.  I can try it and if it works, then great and then we can remove it elsewhere in a separate PR.

Reply:
Ok, I removed it and `clang-format` seems happy.